src/runtime/runtime.cs

```diff
@@ -351,26 +305,23 @@ internal static void Shutdown(ShutdownMode mode)
                     PyEval_SaveThread();
                 }
 
+                ExtensionType.loadedExtensions.Clear();
+                CLRObject.reflectedObjects.Clear();
             }
             else
             {
                 ResetPyMembers();
-                if (mode != ShutdownMode.Extension)
-                {
-                    Py_Finalize();
```

Comment:
This is my only real gripe with this: Embedders would not be able to finalise the Python runtime at all anymore. Maybe we should provide this (`Shutdown()` + `Py_Finalize()`) as a possible footgun with a long name?

Reply:
We can always add it if there is a strong request. Instead what happened is people opening bugs about crashes after restarting runtime because NumPy does not support it. I'd rather avoid it.

Moreover, the behavior of the whole thing when either of runtimes is shut down completely is pretty much undefined.